lib/test_kfence.c

```diff
@@ -31,11 +31,31 @@ int do_test(size_t size)
 	return 0;
 }
 
+int do_test_uaf(size_t size)
+{
+	int i;
+	volatile char *c;
+	void **buffers;
+
+	buffers = kmalloc_array(MAX_ITER, sizeof(void *), GFP_KERNEL);
+	for (i = 0; i < MAX_ITER; i++) {
+		buffers[i] = kmalloc(size, GFP_KERNEL);
+		kfree(buffers[i]);
+		c = ((char *)buffers[i]);
+		(void)*c;
```

Comment:
You could just make this 'READ_ONCE(*buffer)'.

Reply:
Or, if you want to search the whole buffer until the first random "x" appears, something like 'strnstr(buffer, "x", size)'.

Reply:
I like READ_ONCE() more, thanks!
In the future we might want to precisely point at a byte in the buffer to ensure the error report is correct.